src/app/CommandHandler.h

```diff
@@ -59,6 +59,55 @@ class CommandHandler : public Command
         virtual void OnDone(CommandHandler * apCommandObj) = 0;
     };
 
+    class Handle
+    {
+    public:
+        Handle() {}
+        Handle(const Handle & handle) = delete;
+        Handle(Handle && handle)
+        {
+            mpHandler        = handle.mpHandler;
+            mMagic           = handle.mMagic;
+            handle.mpHandler = nullptr;
+            handle.mMagic    = 0;
+        }
+        Handle(decltype(nullptr)) {}
+        Handle(CommandHandler * handle);
+        ~Handle() { Release(); }
```

Comment:
When is a consumer who creates one of these handles _actually_ allowed to delete it? As it stands right now, if a server creates one of these handles on the command callback and destroys it within that call path, it's going to result in the premature transmission of the response, colliding with the logic in `CommandHandler` that _also_ sends the response.

There should be a check in the handle itself that no-ops if it is still in the call path for dispatching a command, preventing premature transmission of the response.

Reply:
This won't happen, and such check is done by always add a refcount and decrease it when finishing dispatching the command.